server/service/src/sync/sync_serde.rs

```diff
@@ -24,6 +24,12 @@ pub fn empty_str_as_option<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
     Ok(Some(T::deserialize(str_d)?))
 }
 
+pub fn ok_or_none<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
```

Comment:
I was gonna call this something confusing like `empty_str_as_option_and_enum_invalid_variant_as_none` however that is obnoxiously long and the real behaviour really just is `ok_or_none`, cause any error will be returned as None.

I had wanted to check specifically for the `invalid enum variant` error that serde throws but couldn't quite figure out how to do so if that is possible.